src/views/UserDashboard.js

```diff
@@ -1,18 +1,59 @@
-import { useContext } from 'react';
+import { useContext, useState, useEffect, useRef } from 'react';
 
 import Summary from './Summary';
 import AddExpenses from './AddExpenses';
 import MonthSelector from '../components/MonthSelector';
 import ExpenseHistory from './ExpenseHistory';
-import ConfirmMonthModal from '../components/Modals/ConfirmMonthModal'
-import ConfirmDeleteModal from '../components/Modals/ConfirmDeleteModal';
+import ChangeMonthModal from '../components/Modals/ChangeMonthModal'
+import DeleteExpenseModal from '../components/Modals/DeleteExpenseModal';
 import EditExpenseModal from '../components/Modals/EditExpenseModal';
-import UserActionAlert from '../components/Alerts/UserActionAlert';
+import Alert from '../components/Alerts/Alert';
 
 import { GlobalContext } from '../store/GlobalState';
+import { getCurrentMonth, getExpenseMonth } from '../utils/helpers';
 
 const UserDashboard = () => {
-  const { isConfirmMonthModalOpen, isConfirmDeleteModalOpen, isEditExpenseModalOpen, isAlertOpen } = useContext(GlobalContext);
+  const currentMonth = getCurrentMonth();
+
+  const { expenses } = useContext(GlobalContext);
+
+  const [selectedMonth, setSelectedMonth] = useState(currentMonth);
+
+  const selectedMonthExpenses = expenses
+    .filter(expense => getExpenseMonth(expense) === selectedMonth)
+    .sort((expenseA, expenseB) => new Date(expenseB.date) - new Date(expenseA.date));
+
+  useEffect(() => {
+    if (selectedMonthExpenses.length === 0) {
+      changeMonthView(currentMonth);
+    }
+  });
+
+  // Modals and alert refs and functions
+  const changeMonthModalRef = useRef(null);
+  const deleteExpenseModalRef = useRef(null);
+  const editExpenseModalRef = useRef(null);
```

Comment:
as discussed on the phone, i am not the best to judge this react solution.
it feels already much better than before.

somehow, i would love to go a step further to have a single variable/ref which contains the current modal and then this gets replaced every time another modal is needed. when there is no modal to show, it is just `null`. i know you can do something `<component :is="someVariable">` which then dynamically decides its contents. however, this still leaves the problem of passing the props. i really don't know how to make this more elegant.